### ISSUE TITLE
Immediate eval in elpaca workflow?

### ISSUE THREAD

**Issue by Lenbok** [progfolio/elpaca#120]
### Elpaca Version

"Elpaca 6c53ce3 HEAD -> master
installer:      0.3
emacs-version:  GNU Emacs 29.0.90 (build 2, x86_64-pc-linux-gnu, GTK+ Version 2.24.33, cairo version 1.16.0)
 of 2023-04-22
git --version:  git version 2.34.1"

### Operating System

Linux

### Description

I've recently converted from `use-package` with `straight.el` to `use-package` with `elpaca`. 

Previously when I wanted to try out a new package, I could place a new `use-package` directive (containing various `:config` etc) in my `init.el` and then `C-x C-e` it to do the install immediately (similarly when updating parts of an existing `use-package` directive). What is the equivalent workflow with elpaca? (`eval`ing probably just enqueues it and I need to somehow "elpaca process"?) It'd be nice if there were a way to just synchronously process when directly `eval`ing the `use-package` declaration.

**Comment by progfolio** [comment]
`eval-last-sexp` should immediately process the order.
Let me know if it doesn't.

**Comment by Lenbok** [comment]
Thanks, looks good so far!

**Comment by dmvianna** [comment]
I was about to make a similar question, but in the context of **developing** a package. I have the git repo locally, which is great, but restarting emacs every time I make a change and want to see the result seems too burdensome. Is there any alternative workload for developing packages within `/elpaca/repos/` except restarting emacs after `M-x elpaca-rebuild`?

**Comment by progfolio** [comment]
> restarting emacs every time I make a change and want to see the result seems too burdensome.

You can develop interactively without restarting Emacs.
However, once you get to a point where you feel like a feature is worth testing, you'd want it to be evaluated in a "fresh" Emacs instance. 
That ensures things are working as expected in a clean, minimal environment.
The simplest approach is to bind a key to launch a new Emacs instance.
However, that's still loading your configuration, which you should aim to keep out of the test as well.
More involved workflows involve setting up tests via a Makefile, or some other tooling.
elpaca-test can be used as well. e.g.

```emacs-lisp
(elpaca-test
   :ref local
   ;;Uncomment if you want to use an interactive session
   ;;:interactive t 
   :init
   (elpaca my-package (my-package-run-tests))
   (elpaca-wait))
```

If the package is simple you may be able to get away with a combination of `unload-feature` and re-loading the package, but I don't recommend that because it doesn't ensure things you've evaluated as you were experimenting are unbound.

**Comment by dmvianna** [comment]
Oh, I just found [alphapapa's dev handbook](https://alphapapa.github.io/emacs-package-dev-handbook/) . Is there something else you recommend? Is this in the inbuilt tutorial?

**Comment by Gleek** [comment]
Hi @progfolio!
It seems that `eval-last-sexp` works as intended after the change. But other eval functions such as `eval-defun` still fail. Packages such as [eros](https://github.com/xiongtx/eros) also don't work correctly.
Both these require a manual execution of `(elpaca-process-queues)` to work correctly.

Do you think these functions should be separately handled or maybe some generic implementation should be done, such as always running process queues once init is complete or some delayed processing mechanism where process queues is run automatically if there is no enqueue after a few milliseconds.

**Comment by Gleek** [comment]
Looks like the implementation for this is already in place and I referenced older code. But it still fails to work for me for some reason sometimes. Will try replicating it again.

**Comment by progfolio** [comment]
> Looks like the implementation for this is already in place and I referenced the
> older code earlier. But it still fails to work for me for some reason sometimes.
> Will try replicating it again.

If you're able to reproduce the issue reliably, please open a separate issue. Thank you.
